design/design-doc-terrajet.md

```diff
@@ -0,0 +1,562 @@
+# Terrajet: Code Generation for Terraform-based Providers
+* Owner: Muvaffak Onus (@muvaf)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
```

Comment:
@muvaf Are you feeling ready to merge this? If so, please change this to approved and merge away.